test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/XPlatPushTests.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NuGet.CommandLine.XPlat;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace NuGet.XPlat.FuncTest
+{
+    public class XPlatPushTests
+    {
+        [Theory]
+        [InlineData(TestServers.MyGet, nameof(TestServers.MyGet), false)]
+        [InlineData(TestServers.ProGet, nameof(TestServers.ProGet), false)]
+        [InlineData(TestServers.Nexus, nameof(TestServers.Nexus), true)]
+        // [InlineData(TestServers.Artifactory, nameof(TestServers.Artifactory), false)] // Trial license expired!
+        // [InlineData(TestServers.Klondike, nameof(TestServers.Klondike), false)] // 500 Internal Server Error pushing
+        // [InlineData(TestServers.NuGetServer, nameof(TestServers.NuGetServer), false)] // doesn't work (credentials?)
+        public async Task PushToServerSucceeds(string sourceUri, string feedName, bool mustDeleteFirst)
+        {
+            // Arrange
+            using (var packageDir = TestFileSystemUtility.CreateRandomTestFolder())
+            {
+                var packageId = "XPlatPushTests.PushToServerSucceeds";
+                var packageVersion = "1.0.0";
+                var packageFile = await TestPackages.GetRuntimePackageAsync(packageDir, packageId, packageVersion);
+                var configFile = XPlatTestUtils.CopyFuncTestConfig(packageDir);
+                var log = new TestCommandOutputLogger();
+
+                var apiKey = XPlatTestUtils.ReadApiKey(feedName);
+                Assert.False(string.IsNullOrEmpty(apiKey));
+
+                if (mustDeleteFirst)
+                {
+                    DeletePackageBeforePush(packageId, packageVersion, sourceUri, apiKey);
+                }
+
+                var pushArgs = new List<string>()
+                {
+                    "push",
+                    packageFile.FullName,
+                    "--source",
+                    sourceUri,
+                    "--api-key",
+                    apiKey
+                };
+
+                // Act
+                int exitCode = Program.MainInternal(pushArgs.ToArray(), log);
+
+                // Assert
+                Assert.Equal(string.Empty, log.ShowErrors());
+                Assert.Equal(0, exitCode);
+                Assert.Contains($@"PUT {sourceUri}", log.ShowMessages());
```

Comment:
Done

---
In reply to: [68083593](https://github.com/NuGet/NuGet.Client/pull/706#discussion_r68083593) [](ancestors = 68083593)